### PR TITLE
Feature/javafx

### PR DESCRIPTION
**Pull request by abstraktor**
# Motivation
- since the current teamcity build workers do not have javafx properly setup any ci test that involved a javafx-editor runs into a deadlock
- The jetbrains-jdk already brings javafx
- For some usecases, some further native libraries are needed

# Usage

The default JDK did not change. If a project wants to use the jetbrains-jdk, it can opt into that by setting env vars.

Set these env vars to use the jetbrains jdk in your shell with current version being: `jbrsdk-8u202-linux-x64-b1483.37`:

```
export JAVA_HOME="/usr/lib/jvm/jbrsdk-8u202-linux-x64-b1483.37/" PATH="/usr/lib/jvm/jbrsdk-8u202-linux-x64-b1483.37/bin:$PATH"
```

# Todos
- [x] download fixed version of jetbrains jdk from jfrog and extract it into `/usr/lib/jvm/<version>/`
- [x] install libgtk2.0-0 and libxslt1.1 to get the missing shared objects
- [x] set `DISPLAY=:0` so that headless X11 knows where the display host is
- [x] shorten wget progress bars to ease scrolling logs
- [x] test if javafx cells are working in editor tests now
- [x] test with Security Analyst 
- [ ] test with some other project (although no interference is expected)

### CODE REVIEW

**Review by wsafonov — CHANGES_REQUESTED** [review]
Just one small thing: please also clean up after downloading the JB JDK, i.e. /tmp/${JBJDK_VERSION}.tar.gz should be removed after unpacking.

**Comment by wsafonov** [comment]
I talked to Prajwal - to unblock issues with JavaFX in SecA project, for now we'll provide a separate agent that would run this patched version of the mbeddr container. Using that agent we can also verify, if other customer/internal projects run smoothly with the JBRJDK. Eventually I would prefer to have one JDK installed on our official mbeddr container. And probably it makes sense to have JBRJDK there, as we are running MPS builds with this container and building languages that likely have to run under exactly that JDK at the end. Let's also discuss this step with @coolya.

**Comment by pngowda** [comment]
There is no TC license left to authorize an extra agent. we'll have to take down one of the existing agents.

**Comment by wsafonov** [comment]
We agreed that for now it won't hurt other projects much, if we temporary take down one of the agents from the default pool and boot up the new agent with the JB JDK for the purpose of testing SecA builds. After completing testing of the new agent for SecA, we can then check builds of other projects if they run on the JB JDK smoothly. 

In parallel, we should discuss pros and cons of switching the default JDK in mbeddr container to JB JDK and if there are any valid reasons why we should not do so. So far we have agreed that because the main purpose of the JDK in the mbeddr docker is to build/test MPS languages, it would really make sense to use the same type of JDK in the build (mainly because of the used JRE for the tests) as the one that would be most probably used to run the final product. One can argue though that some customers could have very specific restrictions as for Java runtime enviroment that can be used to run the products. In such case having JB JDK as default JDK for the build probably would not be ideal.

At the end we may also tend to a decision to provide multiple JDKs on the mbeddr docker to leave it to the project team to decide what JDK type they want to use in the build. In that case we should configure the docker container in the way that a transparent switch between alternative JDK is possible (in this case switching JAVA_HOME value might not be sufficient).

**Comment by abstraktor** [comment]
@coolya I'd vote for finding a way of having SecA and the rest use a shared pool of workers again. The changes this PR does are non-invasive for existing builds. The resulting container just has an additional folder and the java-version will only be used when adding it to the PATH / JAVA_HOME.

So why not have this image serve as worker for all the builds then and let SecA opt into using this jdk?

(All RCPs run JBJDK, right? I'd be surprised, if the other projects would not want to run their tests with their productive JDK. It should be the projects decision though, **when** to switch to it, of course.)

**Pull request merged**
